include/open62541/plugin/eventloop.h

```diff
@@ -311,6 +311,15 @@ struct UA_ConnectionManager {
                       void *application, void *context,
                       UA_ConnectionManager_connectionCallback connectionCallback);
 
+    /* Validate a Connection
+     * ~~~~~~~~~~~~~~~~~
+     * Validate a connection before it is opened. Same parameters are applicable
+     * as for `openConnection`.
+     */
+    UA_StatusCode
+    (*validateConnection)(UA_ConnectionManager *cm,
```

Comment:
What is the advantage of this compared to opening a "real" connection and closing it immediately?
Because you do open an actual connection internally.

Note that DNS lookups can take a lot of time.
So this validation should eventually become asynchronous with a callback - if we not only check the parameters syntactically.
Async DNS is something general for all protocols, not just UDP though...

Reply:
I create a socket internally but I do not bind it and it will not be registered anywhere in the eventloop.
The socket opening is only a way to check that the given connection parameters are valid.

Maybe the function should be called "validateParameters" instead of "validateConnection"?

Anyway it will be put into a separate PR